homeassistant/components/samsungtv/config_flow.py

```diff
@@ -174,3 +177,36 @@ def _try_connect(self):
         self.context["title_placeholders"] = {"model": self._title}
 
         return await self.async_step_confirm()
+
+    @staticmethod
+    @core.callback
+    def async_get_options_flow(config_entry):
+        """Define the config flow to handle options."""
+        return SamsungTVOptionsFlowHandler(config_entry)
+
+
+class SamsungTVOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a SamsungTV options flow."""
+
+    def __init__(self, config_entry):
+        """Initialize."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        if CONF_MAC in self.config_entry.options:
+            current_mac = self.config_entry.options[CONF_MAC]
+        else:
+            hostname = self.config_entry.data[CONF_HOST]
+            try:
+                current_mac = await self.hass.async_add_executor_job(
+                    partial(get_mac_address, **{"hostname": hostname})
+                )
+            except socket.gaierror:
+                current_mac = None
+
+        schema = {vol.Optional(CONF_MAC, default=current_mac): str}
```

Comment:
Is there a cv type for mac address validation? Your current implementation accepts everything if I'm right.

Reply:
Unfortunately there isn't. According to https://github.com/home-assistant/architecture/issues/222 there also might never be.